src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,65 @@
-import React from 'react';
+import React, { ChangeEvent } from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const handleChangeStatus = (event: ChangeEvent<HTMLSelectElement>) => {
+    const statusFromView = event.target.value as Status;
+
+    switch (statusFromView) {
+      case 'all':
+        return dispatch(filterActions.filterAll(statusFromView, query));
+
+      case 'active':
+        return dispatch(filterActions.filterActive(statusFromView, query));
+
+      case 'completed':
+        return dispatch(filterActions.filterCompleted(statusFromView, query));
+
+      default:
+        return dispatch(filterActions.filterAll(statusFromView, query));
+    }
+  };
+
+  const handleChangeQuery = (event: ChangeEvent<HTMLInputElement>) => {
+    const queryFromView = event.target.value;
+
+    switch (status) {
+      case 'all':
+        return dispatch(filterActions.filterAll(Status.ALL, queryFromView));
+      case 'active':
+        return dispatch(filterActions
+          .filterActive(Status.ACTIVE, queryFromView));
+      case 'completed':
+        return dispatch(filterActions
+          .filterCompleted(Status.COMPLETED, queryFromView));
+      default:
+        return dispatch(filterActions.filterAll(Status.ALL, query));
+    }
+  };
```

Comment:
It's overkill. A lot of redundant code.

I'd suggest creating just two action creators: `filterStatus` and `filterQuery`.
They'd look like this: `const filterStatus = (status: Status): FilterStatusAction => ({ type: 'todos/STATUS', payload: status });` and `const filterQuery = (query: string): FilterQueryAction => ({ type: 'todos/QUERY', payload: query });`